src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml

```diff
@@ -49,8 +49,13 @@
 
         <service id="session_listener" class="Symfony\Component\HttpKernel\EventListener\SessionListener">
             <tag name="kernel.event_subscriber" />
-            <argument type="service-locator">
-                <argument key="session" type="service" id="session" on-invalid="null" />
+            <argument type="service">
+                <service class="Symfony\Component\DependencyInjection\ServiceLocator">
+                    <tag name="service_locator" />
```

Comment:
Almost all other tags are of the "X.Y" form. Not sure if we need to call it `container.service_locator`.